### DISCUSSION TITLE
Issue with Energy Conservation for SDF bowl in MuJoCo 3.2.2

### DISCUSSION THREAD

**Discussion by QinOwen** [google-deepmind/mujoco#2286] [Asking for Help]
### Intro

Hi!

I am a student trying to simulate physics problems with MuJoCo. I am using MuJoCo's SDF feature to simulate a ball dropping onto a curved surface and am experiencing unexpected behavior:

1. Energy conservation does not hold in the setup. For example, if a ball is dropped from one side of the bowl, it does not roll to the same height on the other side, even though friction is disabled.

### My setup

•	Operating System: Windows 11
•	MuJoCo Version: 3.2.2

### My question

How can I simulate curved surfaces and ensure that physical properties are accurate?

### Minimal model and/or code that explain my question

XML File Used:

```XML
<mujoco>
  <extension>
    <plugin plugin="mujoco.sdf.bowl">
      <instance name="bowl">
        <config key="height" value="0.0"/>
        <config key="radius" value="1.0"/>
        <config key="thickness" value="0.02"/>
      </instance>
    </plugin>
  </extension>

  <asset>
    <mesh name="bowl">
      <plugin instance="bowl"/>
    </mesh>
  </asset>

  <option sdf_iterations="5" sdf_initpoints="20"/>

  <default>
    <geom solref="0.01 1" solimp=".95 .99 .0001" friction="0 0 0" condim="1"/>
  </default>

  <statistic meansize=".1"/>

  <!-- <include file="scene.xml"/> -->

  <worldbody>

    <light name="left" pos="0 0 4"/>
    <light name="right" pos="-1 -1 2" dir="1 1 -2"/>
    
    <body euler="0 0 0" pos="0 0 1">
      <geom type="sdf" name="bowl" mesh="bowl" rgba=".2 .8 .8 1" condim="1">
        <plugin instance="bowl"/>
      </geom>
    </body>
    <body pos="0.88 0.0 1.0">
      <geom type="sphere" size=".1" rgba=".8 .17 .15 1" condim="1"/>
      <freejoint/>
    </body>

  </worldbody>
</mujoco>


```

I have attached a video as a visual demonstration.
https://github.com/user-attachments/assets/11c7d586-35f3-4523-bc21-9ba9a51316aa

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
Disabling friction does not imply energy conservation. That would only happen with perfectly elastic collisions.

**Comment by QinOwen** [comment]
Thanks for replying. Because I do not want the ball to finally stop at the bottom of the bowl, is there any way that I can enable perfect elastic collisions and prevent the ball from stopping?

Here is a visualization:
![bowl](https://github.com/user-attachments/assets/82ec4b9f-4338-488c-95cd-f0209835fe47)

**Reply by yuvaltassa** [reply]
Not really. You can make perfectly elastic collisions (search the documentation), but then it would bounce rather than roll. I mean, try it and find out. You might get it to work.
